service/service.go

```diff
@@ -19,22 +19,106 @@ package service
 import (
 	"context"
 	"errors"
+	"fmt"
+	"time"
 
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
 	pb "github.com/kubeflow/metadata/api"
 	mlmd "github.com/kubeflow/metadata/ml_metadata"
+	"github.com/kubeflow/metadata/schemaparser"
+)
+
+const (
+	wholeMetaPropertyName string = "__ALL_META__"
+	categoryPropertyName  string = "category"
+	artifactCategory      string = "artifact"
+	defaultTimeout               = 5 * time.Second
 )
 
 // Service implements the gRPC service MetadataService defined in the metadata
 // API spec.
 type Service struct {
-	mlmdClient mlmd.MetadataStoreServiceClient
+	mlmdClient       mlmd.MetadataStoreServiceClient
+	typeNameToMLMDID map[string]int64
+
+	// TODO: Once MLMD type system supports schema annotations, the following two fields can be obtained by quering MLMD types.
+	typeNameToID map[string]string
+	schemaset    *schemaparser.SchemaSet
```

Comment:
I don't think Service needs this. All you need is a separate function, which could be in main like so:

```
func registerPredefinedSchema(ss *schemaParser, schemaRootDir string, mlmdClient *mlmd.MetadatStoreClient) {
...
}
```

This way, Service doesn't need to know about the JSON types. Later, when you move these types to ArtifactType definitions, there's less code to remove in Service. What do you think?

Reply:
For current MLMD, there is no place to store the actual schema for fields. this schema set is the in memory storage for the schemas. so that we can use them to validate fields.

It is can be removed once MLMD can support schema annotations. but now it is needed. That's why I put a TODO above.

Reply:
SGTM, thanks that makes sense.